circle-mlir/circle-mlir/lib/dialect/src/ConstFold.h

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// from tensorflow/compiler/mlir/lite/ir/tfl_ops.cc
+
+#ifndef __CIRCLE_MLIR_DIALECT_CONST_FOLD_H__
+#define __CIRCLE_MLIR_DIALECT_CONST_FOLD_H__
+
+#include "circle-mlir/dialect/CircleDialect.h"
+
+namespace mlir
+{
+namespace Circle
+{
+
+namespace
+{
+
+// Returns new shape with rank 'new_dims' with padded ones on the
+// left if needed.
+inline std::vector<int64_t> GetPaddedShape(ArrayRef<int64_t> old_shape, int new_dims)
+{
+  std::vector<int64_t> new_shape(new_dims, 1);
+  std::copy_backward(old_shape.begin(), old_shape.end(), new_shape.end());
+  return new_shape;
+}
+
+// Helper method that given and 'current_index' representing
+// index in broadcasted tensor, get the index in the flat original tensor.
+// 'shape' is the original shape with padding to match result shape.
+int64_t GetElementIndex(const std::vector<int64_t> &shape,
```

Comment:
It would be nice to add `inline` keyword to prevent multiple definition errors.
```suggestion
inline int64_t GetElementIndex(const std::vector<int64_t> &shape,
```

Reply:
As I wrote above, this file is only included from `CircleDialect.cpp`